src/main/java/liquibase/ext/cassandra/sqlgenerator/TagDatabaseGeneratorCassandra.java

```diff
@@ -0,0 +1,97 @@
+package liquibase.ext.cassandra.sqlgenerator;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import liquibase.Scope;
+import liquibase.database.Database;
+import liquibase.database.ObjectQuotingStrategy;
+import liquibase.datatype.DataTypeFactory;
+import liquibase.ext.cassandra.database.CassandraDatabase;
+import liquibase.sql.Sql;
+import liquibase.sql.UnparsedSql;
+import liquibase.sqlgenerator.SqlGeneratorChain;
+import liquibase.sqlgenerator.core.TagDatabaseGenerator;
+import liquibase.statement.core.TagDatabaseStatement;
+import liquibase.structure.core.Column;
+
+public class TagDatabaseGeneratorCassandra extends TagDatabaseGenerator {
+
+	@Override
+	public int getPriority() {
+		return PRIORITY_DATABASE;
+	}
+
+	@Override
+	public boolean supports(TagDatabaseStatement statement, Database database) {
+		return database instanceof CassandraDatabase;
+	}
+
+	@Override
+	public Sql[] generateSql(TagDatabaseStatement statement, Database database, SqlGeneratorChain sqlGeneratorChain) {
+		ObjectQuotingStrategy currentStrategy = database.getObjectQuotingStrategy();
+		database.setObjectQuotingStrategy(ObjectQuotingStrategy.LEGACY);
+
+		try {
+			String tagEscaped = DataTypeFactory.getInstance().fromObject(statement.getTag(), database).objectToSql(statement.getTag(), database);
+			//Added condition to check database type = cassandra to execute below code
+			if (database instanceof CassandraDatabase) {
```

Comment:
Isn't `getPriority()` and `supports()` enough to ensure we are working with Cassandra DB?

Reply:
In testing with other DB type, found tag command execution failed. Cassandra extension code executed for Postgres and Oracle DB type. So added this IF condition to ensure we are working with Cassandra DB.

Reply:
I think we might need to change `getPriority()` method to be more like in snapshot generators
```
        if (database instanceof CassandraDatabase) {
            return PRIORITY_DATABASE;
        } else {
            return PRIORITY_NONE;
        }
```
but we have same pattern for all sql generators in Cassandra. I think we might leave it as is, and work on issue with pulling wrong generators separately. @nvoxland What's your thoughts?

Reply:
The combination of supports + getPriority that we have for SqlGenerators should be enough to not have this if statement be needed. If you were seeing otherwise, that is something we should look into.

#72 is caused because for the SnapshotGenerator classes, we sort of combined supports + priority into just getPriority() where it returns a negative number if it doesn't support it. https://github.com/liquibase/liquibase-cassandra/blob/e2b3581669264907804488c93202a4122229de20/src/main/java/liquibase/ext/cassandra/snapshot/IndexSnapshotGeneratorCassandra.java#L28 returns the super classes priority , not PRIORITY_NONE like it needs to.

Fixing the issues with an if block in the statement like this will end up causing other issues down the road. We need to make sure we're just using the supports/priority mechanism to either have this class run or pick a different instance.

Reply:
Thanks for your response. I will soon test with these suggestions and raise PR.